pkg/service/auth.go

```diff
@@ -38,11 +39,7 @@ func (self *AuthService) Check(ctx context.Context, req *auth.CheckRequest) (*au
 		return self.deniedResponse(rpc.NOT_FOUND, "Service not found"), nil
 	}
 
-	authContext := AuthContext{
-		ParentContext: &ctx,
-		Request:       req,
-		API:           &apiConfig,
-	}
+	authContext := NewAuthContext(ctx, req, apiConfig)
```

Comment:
Following on https://github.com/3scale-labs/authorino/pull/34/files?file-filters%5B%5D=.go#r567730541...

```suggestion
	authContext := NewAuthContext(ctx, req, apiConfig, log)
```

Reply:
Better to keep logging separated from the auth context